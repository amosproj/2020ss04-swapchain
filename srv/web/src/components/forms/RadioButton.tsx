import React from "react"

type Props = {
    hint?: string
    tag?: string
    name: string
    description?: string
    selected: boolean
    onClick?: () => void
}

export const RadioButton = (props: Props) => {
    let { tag, name, selected, description, hint, onClick } = props
    return (
        <button
            onClick={onClick}
            className={`focus:outline-none flex w-full justify-between border rounded ${selected ? "border-teal-400 shadow" : "bg-gray-100"}`}>
            <div className="flex flex-col items-start p-4">
                <div className="flex items-center justify-between w-full">

                    <span className={`text-left font-bold ${selected ? "text-gray-900" : "text-gray-600"}`}>
                        {name}
                    </span>
                    <span className={`text-xs font-thin ${selected ? "text-gray-600" : "text-gray-400"}`}>{hint}</span>
                </div>
                <p className={`text-sm text-left ${selected ? "text-gray-600" : "text-gray-400"}`}>
                    {description}
                </p>
            </div>
            <div className="flex items-center justify-center">
                <span className={`mr-4 text-3xl font-bold hidden md:block ${selected ? "text-gray-900 " : "text-gray-300"}`}>
                    {tag}
                </span>
            </div>
        </button>

    )

}