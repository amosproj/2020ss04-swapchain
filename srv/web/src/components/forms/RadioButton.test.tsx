import React from 'react';
import { RadioButton } from './RadioButton';
import renderer from 'react-test-renderer';

describe('with all attributes', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<RadioButton name="name" tag="tag" hint="hint" description="description"></RadioButton>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})

describe('without hint', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<RadioButton name="name" tag="tag" description="description"></RadioButton>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})

describe('without tag', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<RadioButton name="name" tag="tag" hint="hint" description="description"></RadioButton>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})

describe('only name and description', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<RadioButton name="name" description="description"></RadioButton>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})