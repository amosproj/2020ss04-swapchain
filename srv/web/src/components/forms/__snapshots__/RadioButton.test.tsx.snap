// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`only name and description renders correctly 1`] = `
<button
  className="focus:outline-none flex w-full justify-between border rounded border-teal-400 shadow hover:bg-white hover:shadow-sm"
>
  <div
    className="flex flex-col items-start justify-between p-4"
  >
    <div
      className="flex items-center justify-between w-full"
    >
      <span
        className="text-left font-bold text-gray-900"
      >
        name
      </span>
      <span
        className="text-xs font-thin text-gray-600"
      />
    </div>
    <p
      className="mt-2 text-sm text-left text-gray-600"
    >
      description
    </p>
  </div>
  <div
    className="flex items-center justify-center"
  >
    <span
      className="mr-4 text-3xl font-bold hidden md:block text-gray-900 "
    />
  </div>
</button>
`;

exports[`with all attributes renders correctly 1`] = `
<button
  className="focus:outline-none flex w-full justify-between border rounded border-teal-400 shadow hover:bg-white hover:shadow-sm"
>
  <div
    className="flex flex-col items-start justify-between p-4"
  >
    <div
      className="flex items-center justify-between w-full"
    >
      <span
        className="text-left font-bold text-gray-900"
      >
        name
      </span>
      <span
        className="text-xs font-thin text-gray-600"
      >
        hint
      </span>
    </div>
    <p
      className="mt-2 text-sm text-left text-gray-600"
    >
      description
    </p>
  </div>
  <div
    className="flex items-center justify-center"
  >
    <span
      className="mr-4 text-3xl font-bold hidden md:block text-gray-900 "
    >
      tag
    </span>
  </div>
</button>
`;

exports[`without hint renders correctly 1`] = `
<button
  className="focus:outline-none flex w-full justify-between border rounded border-teal-400 shadow hover:bg-white hover:shadow-sm"
>
  <div
    className="flex flex-col items-start justify-between p-4"
  >
    <div
      className="flex items-center justify-between w-full"
    >
      <span
        className="text-left font-bold text-gray-900"
      >
        name
      </span>
      <span
        className="text-xs font-thin text-gray-600"
      />
    </div>
    <p
      className="mt-2 text-sm text-left text-gray-600"
    >
      description
    </p>
  </div>
  <div
    className="flex items-center justify-center"
  >
    <span
      className="mr-4 text-3xl font-bold hidden md:block text-gray-900 "
    >
      tag
    </span>
  </div>
</button>
`;

exports[`without tag renders correctly 1`] = `
<button
  className="focus:outline-none flex w-full justify-between border rounded bg-gray-100 hover:bg-white hover:shadow-sm"
>
  <div
    className="flex flex-col items-start justify-between p-4"
  >
    <div
      className="flex items-center justify-between w-full"
    >
      <span
        className="text-left font-bold text-gray-600"
      >
        name
      </span>
      <span
        className="text-xs font-thin text-gray-400"
      >
        hint
      </span>
    </div>
    <p
      className="mt-2 text-sm text-left text-gray-400"
    >
      description
    </p>
  </div>
  <div
    className="flex items-center justify-center"
  >
    <span
      className="mr-4 text-3xl font-bold hidden md:block text-gray-300"
    >
      tag
    </span>
  </div>
</button>
`;
